prog/github/repository_nexus.rb

```diff
@@ -0,0 +1,97 @@
+# frozen_string_literal: true
+
+require_relative "../../lib/util"
+
+class Prog::Github::RepositoryNexus < Prog::Base
+  subject_is :github_repository
+
+  semaphore :destroy
+
+  def self.assemble(installation, name)
+    DB.transaction do
+      repository = GithubRepository.new_with_id(installation_id: installation.id, name: name)
+      repository.skip_auto_validations(:unique) do
+        repository.insert_conflict(target: [:installation_id, :name], update: {last_job_at: Time.now}).save_changes
+      end
+      Strand.new(prog: "Github::RepositoryNexus", label: "wait") { _1.id = repository.id }
+        .insert_conflict(target: :id).save_changes
+    end
+  end
+
+  def client
+    @client ||= Github.installation_client(github_repository.installation.installation_id).tap { _1.auto_paginate = true }
+  end
+
+  def check_queued_jobs
+    queued_runs = client.repository_workflow_runs(github_repository.name, {status: "queued"})[:workflow_runs]
+    Clog.emit("polled queued runs") { {polled_queued_runs: {repository_name: github_repository.name, count: queued_runs.count}} }
+
+    remaining_quota = client.rate_limit.remaining / client.rate_limit.limit.to_f
+    if remaining_quota < 0.1
+      Clog.emit("low remaining quota") { {low_remaining_quota: {repository_name: github_repository.name, limit: client.rate_limit.limit, remaining: client.rate_limit.remaining}} }
+      return (client.rate_limit.resets_at - Time.now).to_i
+    end
+
+    queued_labels = {}
```

Comment:
If you initialize the hash like below, it would automatically set initial values of new keys to 0, so you wouldn't need to set them to 0 yourself.

`queued_labels = Hash.new(0)`

Reply:
Good trick 🚀 I didn't know that.